src/d3a/models/strategy/load_hours.py

```diff
@@ -133,6 +133,12 @@ def event_market_cycle(self):
                 self.energy_requirement_Wh[market.time_slot] = 0.0
                 self.state.desired_energy_Wh[market.time_slot] = 0.0
         self.event_market_cycle_prices()
+        if self.area.current_market:
+            self.state.total_energy_demanded_wh = sum(
+                e for t, e in self.state.desired_energy_Wh.items()
+                if t <= self.area.current_market.time_slot)
```

Comment:
Hmm, unfortunately, I dont see any point in the code, where the self.state.desired_energy_Wh is filled with values. CAn you help me?

Reply:
This file, line 360 https://github.com/gridsingularity/d3a/pull/706/files#diff-5f60b6cdf41a34bdba5921c18ebab298L360

Reply:
Ok, thanks.
Sorry, but why are there multiple entries in the desired_energy_Wh with a time stamp older than the current market?

Reply:
I think you misunderstood the intention here. The problem was that the `total_energy_demanded_Wh` was calculated always using the avg_power_W, which is obviously incorrect for the case of load profiles. In addition to that, the value was not calculated after each market slot, but only at the beginning of the simulation. The self consumption formula though is total_energy_consumed / total_energy_demand (the value we are talking about now). Total energy consumed is aggregated every market slot, while this value always had the final value (precalculated at the beginning of the simulation). This had the very annoying effect IMO that the self-sufficiency and self-consumption values did not display the correct values during the simulation run (because the nominator of the aforementioned division was referring to the values until this point in the simulation, and the denominator for the whole simulation). Therefore in order to sanely debug this I had to also recalculate the `total_energy_demanded_wh` value on every market slot, in order for the results to be correct on every market slot. 
Therefore, I converted the meaning of this value from 'total energy demanded for the whole simulation' to 'total energy demanded until this point in the simulation'.
Hope this clarified things a bit.

Reply:
Ahhhh! Thanks a lot! I confused `total_energy_demanded_wh` with something like `energy_requirement_Wh` here. But we are in the KPIs.
Ok. Your explanation makes sense. Thanks a lot!